packages/web/src/hooks/swap/data/use-swap.tsx

```diff
@@ -20,6 +20,17 @@ export const useSwap = ({ tokenA, tokenB, direction, slippage, swapFee = 15 }: U
   const { account } = useWallet();
   const { swapRouterRepository } = useGnoswapContext();
   const [swapAmount, setSwapAmount] = useState<number | null>(null);
+  const [estimatedLiquidityMax, setEstimatedLiquidityMax] = useState<number | null>(null);
+
+  const shouldFetchData = useCallback(
+    (amount: number | null) => {
+      if (!amount) return false;
```

Comment:
This line returns `false` when `amount === 0`. If `0` is considered a valid value, it would be better to explicitly exclude only null/undefined using `amount === null` instead of `!amount`.

However, since this is related to swap functionality, it's unlikely that `amount` would ever be 0 anyway.

Reply:
@notJoon 
Currently, the `!amount` check is working exactly as intended.
API calls should also not be made when amount is zero, which includes both of the following situations:
1. If 0 is entered as the final value
2. If a zero is temporarily entered while entering a decimal value (for example, ‘0’ → ‘0.’ → ‘0.1’ → ‘0.12’)

Therefore, the current implementation of checking for both null and zero via !amount is appropriate.